node/task.ts

```diff
@@ -725,36 +728,105 @@ export function resolve(...pathSegments: any[]): string {
 export const which = im._which;
 
 /**
- * Returns array of files in the given path, or in current directory if no path provided.  See shelljs.ls
- * @param  {string}   options  Available options: -R (recursive), -A (all files, include files beginning with ., except for . and ..)
+ * Returns array of files in the given path, or in current directory if no path provided.
  * @param  {string[]} paths    Paths to search.
+ * @param  {boolean} recursive Whether to recurse into subdirectories
+ * @param  {boolean} allFiles  Whether to copy all files, include files beginning with ., except for . and ..
  * @return {string[]}          An array of files in the given path(s).
  */
-export function ls(options: string, paths: string[]): string[] {
-    if (options) {
-        return shell.ls(options, paths);
-    } else {
-        return shell.ls(paths);
-    }
+export function ls(paths: string[], recursive: boolean = false, allFiles: boolean = false): string[] {
+    let filePaths: string[] = [];
+    paths.forEach(path => {
+        let command: string = '';
+        if (getPlatform() == Platform.Windows) {
+            command = 'dir ' + path + ' /b';
+            if (recursive) {
+                command += ' /s';
+            }
+            if (allFiles) {
+                command += ' /a';
+            }
+        }
+        else {
+            command = 'ls ' + path;
+            if (recursive) {
+                command += ' -R';
+            }
+            if (allFiles) {
+                command += ' -A';
+            }
+        }
+        const childProcessResponse: string = childProcess.execSync(command).toString();
+        childProcessResponse.split(os.EOL).forEach((file: string) => {
+            if (filePaths.indexOf(file) < 0) {
+                filePaths.push(file);
+            }
+        });
+    });
+
+    return filePaths;
 }
 
 /**
  * Copies a file or folder.
  * 
  * @param     source     source path
  * @param     dest       destination path
- * @param     options    string -r, -f or -rf for recursive and force 
+ * @param     recursive  Copy recursively
+ * @param     force      Force copy
  * @param     continueOnError optional. whether to continue on error
  */
-export function cp(source: string, dest: string, options?: string, continueOnError?: boolean): void {
-    if (options) {
-        shell.cp(options, source, dest);
+export function cp(source: string, dest: string, recursive: boolean = false, force: boolean = false, continueOnError?: boolean): void {
+    try {
+        if (!fs.existsSync(source)) {
```

Comment:
Check dest too?

Reply:
What if dest exists and we are copying a directory.

Reply:
Its fine if dest does or doesn't exist. That's the significance of `force` - it determines if we'll overwrite the file or not. Robocopy handles that for directories no problem, it doesn't do it for files correctly though which is why I have the logical split.